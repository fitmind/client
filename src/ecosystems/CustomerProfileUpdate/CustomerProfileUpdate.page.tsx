import React from 'react';
import { withRouter, RouteComponentProps } from 'react-router';
import { Container, Row, Col, Card, Button, Image, Form } from 'react-bootstrap';
import styled from 'styled-components';
import { ConnectedReduxProps } from '../../redux/reducers/root.reducer';
import { connect } from 'react-redux';
import { ApplicationState } from '../../redux/reducers/root.reducer';
import { customerUserInterface, customerDashboardInterface } from '../../redux/reducers/server-reducer/server.reducer';
import { fetchCustomerDashboardAction } from '../../redux/actions/server/server.actions';
import CONFIG from '../../config/config';

interface PropsFromState {
    customerUser: customerUserInterface;
    customerDashboard: customerDashboardInterface;
}

interface PropsFromDispatch {
    fetchCustomerDashboardAction: typeof fetchCustomerDashboardAction;
}

type CustomerDashboardPageAllProps = PropsFromDispatch & RouteComponentProps & ConnectedReduxProps & PropsFromState;

export class CustomerDashboard extends React.Component<CustomerDashboardPageAllProps> {
    componentDidMount() {
        this.props.fetchCustomerDashboardAction();
    }

    renderUpcomingBookings = () => {
        const rows = [];

        this.props.customerDashboard.upcomingBookings.forEach((booking, index) => {
            rows.push(
                <tr key={booking._id}>
                    <td>{index + 1}</td>
                    <td>{booking.client}</td>
                    <td>{booking.time}</td>
                    <td>{booking.email}</td>
                    <td>{booking.listing}</td>
                    <td>{booking.price}</td>
                </tr>,
            );
        });
        return rows;
    };
    renderPastBookings = () => {
        const rows = [];

        this.props.customerDashboard.pastBookings.forEach((booking, index) => {
            rows.push(
                <tr key={booking._id}>
                    <td>{index + 1}</td>
                    <td>{booking.client}</td>
                    <td>{booking.time}</td>
                    <td>{booking.email}</td>
                    <td>{booking.listing}</td>
                    <td>{booking.price}</td>
                </tr>,
            );
        });
        return rows;
    };

    render() {
        const { customerUser } = this.props;
        return (
            <Container fluid>
                <Row>
                    <Col md={4} />
                    <Col md={4}>
                        <CardWrapper>
                            <Card>
                                <Card.Header as="h5">Edit your Profile</Card.Header>
                                <Card.Body>
                                    <CenterContainer>
                                        <Image src={customerUser.phoneUrl} rounded />
                                        <Form>
                                            <Form.Group>
                                                <Form.Label>Description of yourself *</Form.Label>
                                                <Form.Control
                                                    as="textarea"
                                                    rows="6"
                                                    defaultValue={customerUser.description}
                                                />
                                            </Form.Group>
                                            <Form.Group>
                                                <Form.Label>Phone number *</Form.Label>
                                                <Form.Control
                                                    type="text"
                                                    placeholder="Enter phone"
                                                    defaultValue={customerUser.phone}
                                                />
                                            </Form.Group>

                                            <Form.Group>
                                                <Form.Label>Coaches you are looking for *</Form.Label>
                                                <Form.Control
                                                    as="select"
                                                    multiple
                                                    defaultValue={customerUser.interestedInExperiseAreas}
                                                >
                                                    {Object.keys(CONFIG.expertises).map((key: string) => (
                                                        <option key={key} value={CONFIG.expertises[key].value}>
                                                            {CONFIG.expertises[key].display}
                                                        </option>
                                                    ))}
                                                </Form.Control>
                                            </Form.Group>
                                            <Button variant="primary">Update Profile</Button>
                                        </Form>
                                    </CenterContainer>
                                </Card.Body>
                            </Card>
                        </CardWrapper>
                    </Col>
                </Row>
            </Container>
        );
    }
}

const CenterContainer = styled.div`
    text-align: center;
`;

const CardWrapper = styled.div`
    margin-top: 5rem;
`;

const mapDispatchToProps = {
    fetchCustomerDashboardAction,
};
const mapStateToProps = (state: ApplicationState) => ({
    customerUser: state.server.customerUser,
    customerDashboard: state.server.customerDashboard,
});
export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps,
    )(CustomerDashboard),
);
