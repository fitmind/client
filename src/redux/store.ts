import { Store, createStore, applyMiddleware } from 'redux';
import { ApplicationState, storeInitialState, createRootReducer } from './reducers/root.reducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import { rootSaga } from './saga/root.saga';
import { History, createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

// We use hash history because this example is going to be hosted statically.
// Normally you would use browser history.
export const history: History = createBrowserHistory();

export const configureStore = (initialState: ApplicationState): Store<ApplicationState> => {
    const sagaMiddleware = createSagaMiddleware();
    const composeEnhancers = composeWithDevTools({});
    let store = createStore(
        createRootReducer(history),
        initialState,
        composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware)),
    );

    sagaMiddleware.run(rootSaga);
    return store;
};

const store: Store<ApplicationState> = configureStore(storeInitialState);

export default store;
