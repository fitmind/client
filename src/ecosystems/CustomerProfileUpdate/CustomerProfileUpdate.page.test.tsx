import React from 'react';
import { shallow } from 'enzyme';
import CustomerProfileUpdate from './CustomerProfileUpdate.page';
import { Provider } from 'react-redux';
import store, { history } from '../../redux/store';
import { Router } from 'react-router';

describe('Customer Profile Update page', () => {
    it('renders without crashing', () => {
        shallow(
            <Provider store={store}>
                <Router history={history}>
                    // @ts-ignore
                    <CustomerProfileUpdate />
                </Router>
            </Provider>,
        );
    });
});
