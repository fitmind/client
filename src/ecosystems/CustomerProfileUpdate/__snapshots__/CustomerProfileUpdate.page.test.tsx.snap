// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Profile Update page should match the snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    />
    <div
      className="col-md-4"
    >
      <div
        className="sc-bwzfXH kUNjUS"
      >
        <div
          className="card"
        >
          <h5
            className="card-header"
          >
            Edit your Profile
          </h5>
          <div
            className="card-body"
          >
            <div
              className="sc-bdVaJa kmXoBq"
            >
              <img
                className="rounded"
              />
              <form
                className=""
              >
                <div
                  className="form-group"
                >
                  <label
                    className="form-label"
                  >
                    Description of yourself *
                  </label>
                  <textarea
                    className="form-control"
                    rows="6"
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label
                    className="form-label"
                  >
                    Phone number *
                  </label>
                  <input
                    className="form-control"
                    placeholder="Enter phone"
                    type="text"
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label
                    className="form-label"
                  >
                    Coaches you are looking for *
                  </label>
                  <select
                    className="form-control"
                    multiple={true}
                  >
                    <option
                      value="YOGA_COACH"
                    >
                      Yoga Teacher
                    </option>
                    <option
                      value="PERSONAL_COACH"
                    >
                      Personal Trainer
                    </option>
                    <option
                      value="NUTRITIONIST"
                    >
                      Nutritionist
                    </option>
                    <option
                      value="LIFE_COACH"
                    >
                      Life Coach
                    </option>
                  </select>
                </div>
                <button
                  className="btn btn-primary"
                  disabled={false}
                  type="button"
                >
                  Update Profile
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
